### PR TITLE
Refactor flag into flexbox, add reverse

### LINKED ISSUE #2907: Re-write the Flag object to use Flexbox instead of table
The Flag Object uses `display: table` to allow the content to the right of the media to be vertically positioned. This should be re-written to use Flexbox instead.

This will also resolve https://github.com/sparkdesignsystem/spark-design-system/issues/1139

**AC:**
- HTML Flag object will use flexbox

### PR DESCRIPTION
**Pull request by RVMendoza**
## What does this PR do?
HTML - Refactor flag to use flexbox, and add reverse story
flag

### Associated Issue
Fixes #2907 

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in HTML
 - [x] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: html/objects/_flag.scss [hunk 1]
```diff
@@ -7,45 +7,28 @@
...
-  display: table;
-  width: 100%;
-}
-
-.sprk-o-Flag__body,
-.sprk-o-Flag__figure {
-  display: table-cell;
-  vertical-align: top;
+  display: flex;
+  align-items: flex-start;
```
Comment (afebbraro) [inline html/objects/_flag.scss]: Why is the flex-start needed here?

File: html/objects/_flag.scss [hunk 2]
```diff
@@ -7,45 +7,28 @@
...
-
-  > img {
-    display: block;
-    max-width: none;
-  }
 }
 
 .sprk-o-Flag__body {
-  width: 100%;
+  flex: 1;
```
Comment (afebbraro) [inline html/objects/_flag.scss]: flex: 1, the shorthand, has issues in ie11 sometimes. Can you take a look there.

File: html/objects/_flag.scss [hunk 3]
```diff
@@ -7,45 +7,28 @@
...
 .sprk-o-Flag--rev {
-  direction: rtl;
-
-  > .sprk-o-Flag__body,
-  > .sprk-o-Flag__figure {
-    direction: ltr;
-  }
-
   > .sprk-o-Flag__figure {
+    order: 1;
```
Comment (afebbraro) [inline html/objects/_flag.scss]: What were your thoughts on using the alignment properties vs. using order?

File: html/objects/_flag.scss [hunk 4]
```diff
@@ -132,15 +115,15 @@
 }
 
 @media all and (max-width: $sprk-flag-stacked-breakpoint) {
-  
+
   /// Stacked flag at smaller screen width.
   .sprk-o-Flag--stacked {
-    direction: ltr;
+    flex-flow: column;
```
Comment (afebbraro) [inline html/objects/_flag.scss]: If we are only setting column, we dont need the whole shorthand `flex-flow`. Can just just `flex-direction`

Comment (RVMendoza) [reply html/objects/_flag.scss]: You're right!

File: html/objects/_flag.scss (hunk 3 above)
Comment (RVMendoza) [reply html/objects/_flag.scss]: I changed it to flex-direction reverse-row -- it feels easier to understand

File: html/objects/_flag.scss (hunk 1 above)
Comment (RVMendoza) [reply html/objects/_flag.scss]: you're right, it's not necessary, it's default is flex stretch if not provided and that would be fine for this.

File: src/pages/using-spark/components/flag.mdx [hunk 5]
```diff
@@ -38,6 +38,13 @@ area can be set to any Spacing value.
 - On small viewports, the content area
 can be configured to move below the media.
 
+### Reverse Modifier
```
Comment (afebbraro) [inline src/pages/using-spark/components/flag.mdx]: Gonna message you about adding something in here about a11y with reversing the order of the elements.

Comment (RVMendoza) [reply src/pages/using-spark/components/flag.mdx]: image

File: html/objects/_flag.scss [hunk 6]
```diff
@@ -132,15 +117,15 @@
...
-    direction: ltr;
+    flex-direction: column;
 
-    > .sprk-o-Flag__body,
-    > .sprk-o-Flag__figure,
-    & {
-      display: block;
+    &.sprk-o-Flag--rev {
+      > .sprk-o-Flag__figure {
+        order: 0;
```
Comment (afebbraro) [inline html/objects/_flag.scss]: hey @RVMendoza , Did you forget to remove this order property? Looks like it's not doing anything.

**Pull request merged**